src/NBT.jl

```diff
@@ -0,0 +1,215 @@
+# ParseNBT files and import/export .schematic files
+#
+# NBT is a file format used by Minecraft to store data. https://minecraft.gamepedia.com/NBT_format
+# A specification was provided by Notch which can be accessed using archive.org
+# http://web.archive.org/web/20110723210920/http://www.minecraft.net/docs/NBT.txt
+#
+# The format of a Named Tag is :
+#     byte tagType
+#     TAG_String name
+#     [payload]
+#
+# [payload] depends on the type of Named Tag.
+# Name Tags are Named Tags without the name, [payload] is the same
+#
+# NBT files are mostly GZip Compressed however there are places where unzipped
+# files are used.
+# Use the GZip.jl Package to read/write GZip streams: https://github.com/JuliaIO/GZip.jl
+#
+# We don't differentiate between Named and Nameless files. Nameless Tags are
+# Named Tags with an empty string as their name.
+
+abstract type Tag end
```

Comment:
If we export this name, then Tag is too common a name. Call it NBTTag or MCTag.

Reply:
Will `NamedTag` be fine ?

Reply:
Sure